data-science-onramp/data-ingestion/setup.py

```diff
@@ -0,0 +1,149 @@
+from random import choice, choices, randint, seed
+import sys
+
+from time import time_ns
+
+from google.cloud import bigquery
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+
+from pyspark.sql.functions import UserDefinedFunction
+from pyspark.sql.types import IntegerType, StringType
+
+
+# Create a SparkSession under the name "setup". Viewable via the Spark UI
+spark = SparkSession.builder.appName("setup").getOrCreate()
+
+bucket_name = sys.argv[1]
+upload = True  # Whether to upload data to BigQuery
+
+# Check whether or not results should be uploaded
+try:
+    sys.argv[2]
+    upload = False
+except IndexError:
+    print("Results will be uploaded to BigQuery")
+
+table = "bigquery-public-data.new_york_citibike.citibike_trips"
+
+# Check if table exists
+try:
+    df = spark.read.format('bigquery').option('table', table).load()
+except Py4JJavaError:
+    print(f"{table} does not exist. ")
+    sys.exit(0)
+
+# START MAKING DATA DIRTY
+
+
+def random_select(items, cum_weights):
+    '''Picks an item according to the cumulative weights'''
+    return choices(items, cum_weights=cum_weights, k=1)[0]
+
+
+def tripduration(duration):
+    '''Converts trip duration to other units'''
+    seconds = str(duration) + " s"
+    minutes = str(float(duration) / 60) + " min"
+    hours = str(float(duration) / 3600) + " h"
+    return random_select([seconds, minutes, hours, str(randint(-1000, -1))],
+                         [0.3, 0.6, 0.9, 1])
+
+
+def station_name(name):
+    '''Replaces '&' with '/' with a 50% chance'''
+    return choice([name, name.replace("&", "/")])
+
+
+def usertype(user):
+    '''Manipulates the user type string'''
+    return choice([user, user.upper(), user.lower(),
+                  "sub" if user == "Subscriber" else user,
+                   "cust" if user == "Customer" else user])
+
+
+def gender(s):
+    '''Manipulates the gender string'''
+    return choice([s, s.upper(), s.lower(),
+                  s[0] if len(s) > 0 else "",
+                   s[0].lower() if len(s) > 0 else ""])
```

Comment:
Check the formatting here, should all be lined up.

Reply:
running `black` on everything will be a fast format fix